src/ott/neural/solvers/base_solver.py

```diff
@@ -0,0 +1,485 @@
+# Copyright OTT-JAX
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from abc import ABC, abstractmethod
+from pathlib import Path
+from types import MappingProxyType
+from typing import Any, Callable, Dict, Literal, Mapping, Optional, Tuple, Union
+
+import jax
+import jax.numpy as jnp
+
+import optax
+from flax.training import train_state
+
+from ott.geometry import costs, pointcloud
+from ott.geometry.pointcloud import PointCloud
+from ott.neural.models import models
+from ott.problems.linear import linear_problem
+from ott.problems.quadratic import quadratic_problem
+from ott.solvers.linear import sinkhorn
+
+
+class BaseNeuralSolver(ABC):
```

Comment:
Right now, this base class also does not seem to provide any benefit (should be more appropriate be called a trainer maybe).
Would consider removing from now, or think about how to make it a bit more useful rather than having just a common supertype.

Reply:
I think later we might have a common `save/load` method.

In general, the main concern about the current/previous ott-neural models were that they had completely different implementation ideas, and hence - tbh - I think having a base class which ensures similar use might be a good guideline.

But it's your call of course.